ext/datadog_profiling_native_extension/heap_recorder.c

```diff
@@ -933,16 +858,15 @@ heap_stack* heap_stack_new(ddog_prof_Slice_Location locations) {
   return stack;
 }
 
-void heap_stack_free(heap_stack *stack) {
-  for (uint64_t i = 0; i < stack->frames_len; i++) {
-    heap_frame *frame = &stack->frames[i];
-    ruby_xfree(frame->name);
-    ruby_xfree(frame->filename);
+void heap_stack_free(heap_recorder *recorder, heap_stack *stack) {
+  for (u_int16_t i = 0; i < stack->frames_len; i++) {
+    unintern_or_raise(recorder, stack->frames[i].filename);
+    unintern_or_raise(recorder, stack->frames[i].name);
   }
```

Comment:
Minor: More of a naming thing, but I had to check exactly what `unintern` does on the libdatadog side, since I was thinking "wait, does this mean we're deleting the string right now?"

Maybe release or something like that may be a bit more explicit? (Perhaps even decrement_usage?)